### PR TITLE
Fix some small imperfections in the API.

### LINKED ISSUE #609: Do a last review pass on the libzim api before releasing.
Libzim7 changes a lot of its API and those changes come from several big PR.
It would be nice to have a last global look to check for API consistency before doing the release.

### PR DESCRIPTION
**Pull request by mgautierfr**
Fix #609

### CODE REVIEW

File: src/cluster.cpp [hunk 1]
```diff
@@ -49,10 +49,15 @@ getClusterReader(const Reader& zimReader, offset_t offset, CompressionType* comp
 {
   uint8_t clusterInfo = zimReader.read(offset);
   // Very old zim files used 0 as a "default" compression, which means no compression.
-  if(clusterInfo&0x0F == 0) {
+  uint8_t compInfo = clusterInfo&0x0F;
```
Comment (maneeshpm) [inline src/cluster.cpp:52]: A space before and after the & operator can improve readability

**Review by maneeshpm — APPROVED** [review]

**Review by veloman-yunkan — APPROVED** [review]
LGTM

File: src/cluster.cpp (hunk 1 above)
Comment (mgautierfr) [reply src/cluster.cpp]: Done

**Comment by mgautierfr** [comment]
Last 4 commits fix review comment of Maneesh and fix other things.

They are pretty easy.

**Pull request merged**